public/API/API_GetGameFromHash.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/**
+ *  API_GetGameFromHash - looks up a game from a given hash string
+ *    h : game hash
+ *
+ *  int        ID                         id of the game
+ *  string     Title                      name of the game
+ *  int        ConsoleID                  unique identifier of the console associated to the game
+ *  string     ConsoleName                name of the console associated to the game
+ */
+
+use App\Platform\Models\GameHash;
+use App\Platform\Models\System;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Validator;
+
+$input = Validator::validate(Arr::wrap(request()->query()), [
+    'h' => ['required', 'size:32'],
+]);
+
+$givenHash = request()->query('h');
+
+$foundHash = GameHash::with('game')->where('MD5', $givenHash)->first();
+
+if (!$foundHash) {
+    return response()->json(['message' => 'Could not find a game matching the given hash.'], 400);
+}
+
+$foundSystem = System::find($foundHash->game->ConsoleID);
+
+return response()->json([
+    'ID' => $foundHash->game->ID,
+    'Title' => $foundHash->game->Title,
+    'ConsoleID' => $foundHash->game->ConsoleID,
+    'ConsoleName' => $foundSystem?->Name ?? null,
```

Comment:
Would it be worth returning the hash description?
```
'Description' => $foundHash->name,
```

```
{
    "ID":1,
    "Title":"Sonic the Hedgehog",
    "Description":"Sonic The Hedgehog (USA, Europe).md",
    "ConsoleID":1,
    "ConsoleName":"Mega Drive"
}
```

Reply:
Probably so, added in https://github.com/RetroAchievements/RAWeb/pull/2090/commits/8ec90f1122c9f8f7847bdfdebacc32a8be1e9d4a.